aws_lambda_builders/__init__.py

```diff
@@ -2,4 +2,4 @@
 AWS Lambda Builder Library
 """
 __version__ = '0.0.5'
-RPC_PROTOCOL_VERSION = "0.1"
+RPC_PROTOCOL_VERSION = "0.2"
```

Comment:
I don't think we want to bump this right now. Since we release aws-lambda-builders and build it into a docker image before we release SAM CLI, this breaks all customers that are not using `--skip-pull-image` for all runtimes `sam build` supports. SAM CLI does not have a mechanism to pin to a docker image, so customers are left hanging until we do a SAM CLI release. Until we have a good mechanism to release this package without impacting customers using SAM CLI or releasing together, we can't update this version.

Reply:
will revise the version back to 0.1, we need a plan for updating the protocol version. I'll capture that as a github issue.

Reply:
Actually not. 

There are two versions at play here 
- Builder library version within SAM CLI. Think of this as a client
- Buildre library within the container. Think of this as the server

Client issues a `build` RPC command to server.

SAM CLI Builder (client) |  Builder Version in Container (Server) | Compatibility
---------|-----------------|--------------
0.1 | 0.2 | YES - New version of server is backwards compat with old version of client
0.2 | 0.1 | NO - New client is not compatible with old server

We auto-upgrade Docker containers. So if someone gets newer version of container aka server, SAM CLI will continue to work. But if someone upgrades SAM CLI but uses --skip-pull-image to stick with older container version, their existing builds will break. This is because newer client has behavior that older servers don't support.

The thought is, we will try to be back compat as much as possible. So checking if `client version <= server version` is good enough. At some point we will intentionally break compat when we will rev major version.

Does this make sense?

Reply:
@sanathkr Thanks for the detailed explanation. I had a miss understanding of the SAM CLI interactions.